### PR TITLE
Handle collisions in transactions table

### LINKED ISSUE #84: Some rows are missing because of transaction hash collisions
In `accounts` table, we have implicit account that still should exist `ae384ca1c2c93a4029e34fb52a5f72adebc5ff10b8b35fdd628ab2e7c874d088`

But we do not have it in RPC, here is the request body
```
{
  "jsonrpc": "2.0",
  "id": "dontcare",
  "method": "query",
  "params": {
    "request_type": "view_account",
    "block_id": 33039470,
    "account_id": "ae384ca1c2c93a4029e34fb52a5f72adebc5ff10b8b35fdd628ab2e7c874d088"
  }
}
```

### PR DESCRIPTION
**Pull request by telezhnaya**
This PR will hopefully close #84 in one day

### CODE REVIEW

File: src/db_adapters/receipts.rs [hunk 1]
```diff
@@ -13,16 +14,38 @@ use crate::schema;
...
+async fn store_chunk_receipts(
     pool: &actix_diesel::Database<PgConnection>,
     receipts: &[near_indexer::near_primitives::views::ReceiptView],
-    block_hash: &str,
+    block_hash: &near_indexer::near_primitives::hash::CryptoHash,
     chunk_hash: &near_indexer::near_primitives::hash::CryptoHash,
     block_timestamp: u64,
     strict_mode: bool,
 ) -> anyhow::Result<()> {
-    if receipts.is_empty() {
```
Comment (khorolets) [inline src/db_adapters/receipts.rs:23]: Why did you remove this?

File: src/db_adapters/transactions.rs [hunk 2]
```diff
@@ -1,22 +1,56 @@
...
+    if inserted_receipt_ids.len() == tried_to_insert_transactions_count {
+        return Ok(());
     }
 
-    let futures = shards
+    // https://github.com/near/near-indexer-for-explorer/issues/84
+    // TLDR: it's the hack to store transactions with collided hashes
+    // It should not happen, but unfortunately it did,
+    // we have ~10 such transactions in Mainnet for now
+    let transaction_hash_suffix = "_issue84_".to_owned() + &block_height.to_string();
```
Comment (khorolets) [inline src/db_adapters/transactions.rs:51]: It looks more like a prefix

File: src/db_adapters/transactions.rs [hunk 3]
```diff
@@ -25,33 +59,61 @@ pub(crate) async fn store_transactions(
                 chunk
                     .transactions
                     .iter()
-                    .collect::<Vec<&near_indexer::IndexerTransactionWithOutcome>>(),
+                    .enumerate()
```
Comment (khorolets) [inline src/db_adapters/transactions.rs:62]: do we need this `enumerate` here? I don't see the use, could you point me where to look if I'm wrong?

**Review by khorolets — APPROVED** [review]

File: src/db_adapters/receipts.rs (hunk 1 above)
Comment (telezhnaya) [reply src/db_adapters/receipts.rs:23]: I moved this check into line 26 (`filter`)

File: src/db_adapters/transactions.rs (hunk 3 above)
Comment (telezhnaya) [reply src/db_adapters/transactions.rs:62]: I had to put it here because we need to pass indexes for collided transactions, they will not be sequential

File: src/main.rs [hunk 4]
```diff
@@ -42,29 +42,30 @@ async fn handle_message(
...
+        streamer_message.block.header.timestamp,
+        strict_mode,
+    );
+
+    // We can process transactions and receipts in parallel
+    // because most of receipts depend on transactions from previous blocks,
+    // so we can save up some time here.
+    // In case of local receipts (they are stored in the same block with corresponding transaction),
+    // we hope retry logic will cover it fine
+    try_join!(transactions_future, receipts_future)?;
```
Comment (telezhnaya) [inline src/main.rs:68]: @frol I think we want to put this change in a separate PR and test the performance separately

Comment (khorolets) [reply src/main.rs:68]: I'd suggest including it here. We can deploy on separate testnet node to check how's the performance and revert it if necessary

File: src/db_adapters/transactions.rs (hunk 2 above)
Comment (frol) [reply src/db_adapters/transactions.rs:51]: We craft a prefix here to pass it into the store_chunk_transactions function below

**Review by frol — APPROVED** [review]

**Comment by telezhnaya** [comment]
I want to save the reference here
https://github.com/near/near-indexer-for-explorer/blob/master/src/db_adapters/transactions.rs#L53-L57

```rust
let transaction_hash_suffix = "_issue84_".to_owned() + &block_height.to_string();
```
